graylog2-web-interface/src/components/graylog/Panel.jsx

```diff
@@ -7,59 +9,220 @@ import teinte from 'theme/teinte';
-        border-bottom-color: ${borderColor};
-      }
+
+    .badge {
+      color: ${backgroundColor(hex)};
+      background-color: ${hex};
     }
-  `;
-};
+  }
+
+  & > ${PanelFooter} {
+    + .panel-collapse > .panel-body {
+      border-bottom-color: ${borderColor(hex)};
+    }
+  }
+`;
 
 const StyledPanel = styled(BootstrapPanel)`
   background-color: ${teinte.primary.due};
 
+  .panel-group {
+    ${PanelHeading} {
+      + .panel-collapse > .panel-body,
+      + .panel-collapse > .list-group {
+        border-top-color: ${teinte.secondary.due};
+      }
+    }
+
+    ${PanelFooter} {
+      + .panel-collapse .panel-body {
+        border-bottom-color: ${teinte.secondary.due};
+      }
+    }
+  }
+
+  ${bsStyleThemeVariant(panelVariantStyles)};
+`;
+
+const deprecatedVariantStyles = hex => css`
+  /** NOTE: Deprecated & should be removed in 4.0 */
+  border-color: ${borderColor(hex)};
+
+  & > .panel-heading {
+    color: ${colorLevel(backgroundColor(hex), 9)};
+    background-color: ${backgroundColor(hex)};
+    border-color: ${borderColor(hex)};
+
+    + .panel-collapse > .panel-body {
+      border-top-color: ${borderColor(hex)};
+    }
+    .badge {
+      color: ${backgroundColor(hex)};
+      background-color: ${hex};
+    }
+  }
+
+  & > .panel-footer {
+    + .panel-collapse > .panel-body {
+      border-bottom-color: ${borderColor(hex)};
+    }
+  }
+`;
+
+const DeprecatedStyledPanel = styled(BootstrapPanel)`
+  /** NOTE: Deprecated & should be removed in 4.0 */
+  background-color: ${teinte.primary.due};
 
   .panel-footer {
     background-color: ${teinte.secondary.tre};
     border-top-color: ${teinte.secondary.due};
   }
-
   .panel-group {
     .panel-heading {
       + .panel-collapse > .panel-body,
       + .panel-collapse > .list-group {
         border-top-color: ${teinte.secondary.due};
       }
     }
-
     .panel-footer {
       + .panel-collapse .panel-body {
         border-bottom-color: ${teinte.secondary.due};
       }
     }
   }
 
-  ${bsStyleThemeVariant(panelVariantStyles)}
+  ${bsStyleThemeVariant(deprecatedVariantStyles)}
 `;
 
-export default StyledPanel;
+const Panel = ({
+  title,
+  children,
+  collapsible,
+  defaultExpanded,
+  expanded,
+  footer,
+  header,
+  onToggle,
+  ...props
+}) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  React.useEffect(() => {
+    setIsExpanded((defaultExpanded && expanded)
+      || (!defaultExpanded && expanded)
+      || (defaultExpanded && isExpanded === expanded));
+  }, [expanded]);
+
+  const handleToggle = (nextIsExpanded) => {
+    setIsExpanded(nextIsExpanded);
+    onToggle(nextIsExpanded);
+  };
+
+  if (header || footer || title || collapsible || typeof children === 'string') {
+    /** NOTE: Deprecated & should be removed in 4.0 */
+    /* eslint-disable-next-line no-console */
+    console.warn('Graylog Notice: ', 'You have used a deprecated `Panel` prop, please check the documentation to use the latest `Panel`.');
```

Comment:
Console warnings log on every render and are very noisy, making it hard to see any other errors or warnings in the page. Could we only render them once, maybe using `useEffect`?